WcaOnRails/lib/results_validators/competitions_results_validator.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module ResultsValidators
+  class CompetitionsResultsValidator < GenericValidator
+    @@desc = "This validator is an aggregate of an arbitrary set of other validators, running on an arbitrary set of competitions."
+
+    attr_reader :results, :persons
+    ALL_VALIDATORS = [
+      EventsRoundsValidator,
+      PositionsValidator,
+      IndividualResultsValidator,
+      ScramblesValidator,
+      CompetitorLimitValidator,
+      AdvancementConditionsValidator,
+      PersonsValidator,
+    ].freeze
+
+    # Takes a list of validator classes, and if it should process real results
+    # or not.
+    def initialize(validators: [], check_real_results: false)
+      super()
+      # If no validator is given, assume we should apply all.
+      @validators = validators.empty? ? ALL_VALIDATORS : validators
+      @check_real_results = check_real_results
+      @results = []
+      @persons = []
+    end
+
+    def check_real_results?
+      @check_real_results
+    end
+
+    def has_results?
+      @results.any?
+    end
+
+    def persons_by_id
```

Comment:
Is this used somewhere? Did a quick look and couldn't find where.

Reply:
Yes here:
https://github.com/thewca/worldcubeassociation.org/blob/f58645791ae44da326d839b6f098be7dd4b6f0d5/WcaOnRails/app/views/results_submission/_table_for_round.html.erb#L2

I definitely don't quite like that, but I couldn't find an alternative where I don't rework the whole thing or duplicate the code.
I do plan on revisiting how the validator works a bit, just not in that PR :p